app/routes.py

```diff
@@ -1,2 +1,187 @@
+
+@task_list_bp.route("/<task_id>/mark_complete", methods = ["PATCH"], strict_slashes=False)
+def update_completed_task(task_id):
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return make_response(" ", 404)
+    
+    url = "https://slack.com/api/chat.postMessage"
+    data = {
+        "channel": "C020ZEDG7AS",
+        "text": f"Someone just completed the {task.title}"
+    }
+
+    headers = {
+        "Authorization": f"Bearer {slack_key}"
+    }
+    slack_response = requests.post(url, data=data, headers=headers)
+
+    task.completed_at = datetime.now()
+    db.session.commit()
+
+    db.session.commit()
+    return make_response(jsonify({"task": task.to_json()}), 200)
+
+@task_list_bp.route("/<task_id>/mark_incomplete", methods = ["PATCH"], strict_slashes=False)
+def update_incompleted_task(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response(" ", 404)
+
+    task.completed_at = None
+    db.session.commit()
+
+    return make_response(jsonify({"task": task.to_json()}), 200)
+
+@goal_bp.route("", methods = ["POST", "GET"])
+def handle_goal():
+    if request.method == "POST":
+        request_body = request.get_json()
+
+        if "title" not in request_body:
+            return make_response(jsonify({"details": f"Invalid data" }), 400)
+    
+        new_goal = Goal(title=request_body["title"])
+        
+        db.session.add(new_goal)
+        db.session.commit()
+
+        return make_response({
+            "goal": new_goal.goal_json()}, 201)
+
+    elif request.method == "GET":
+        goals = Goal.query.all()
+
+        goal_list = []
+        for goal in goals:
+            goal_list.append(goal.goal_json())
+        return make_response(jsonify(goal_list))
+
+@goal_bp.route("/<goal_id>", methods = ["GET", "PUT", "DELETE"])
+def deal_w_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+
+    if goal is None:
+        return make_response(" ", 404)
+
+    if request.method == "GET":
+        return make_response(jsonify({
+            "goal": goal.goal_json()
+        }))
+
+    elif request.method == "PUT":
+        form_data = request.get_json()
+        goal.title = form_data["title"]
+
+        db.session.commit()
+        return make_response(jsonify({"goal": goal.goal_json()}))
+
+    elif request.method == "DELETE":
+        db.session.delete(goal)
+        db.session.commit()
+
+        return make_response({
+            "details": f'Goal {goal.goal_id} "{goal.title}" successfully deleted'})
+
+@goal_bp.route("/<goal_id>/tasks", methods = ["POST", "GET"])
+def goal_by_task(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal is None:
+        return make_response(" ", 404)
+
+    if request.method == "POST":
+        request_body = request.get_json()
+        form_data = {
+            "goal_id": goal_id
+        }
+        for task_id in request_body["task_ids"]:
+            handle_task_helper(task_id, "PUT", form_data)
```

Comment:
This is the only place that handle_task_helper is used other than line 52.  I might split out the 'PUT' functionality into a helper function, and then call it here and from the function that handles the route on line 50, rather than having a one line function at line 51.